src/cluster_asm.c

```diff
@@ -2847,24 +2865,35 @@ int clusterAsmProcess(const char *task_id, int event, void *arg, char **err) {
     if (err) *err = NULL;
 
     switch (event) {
-        case ASM_EVENT_IMPORT_START:
-            ret = asmCreateImportTask(task_id, arg, &errsds) ? C_OK : C_ERR;
+        case ASM_EVENT_IMPORT_START: {
+            /* Validate the slot ranges. */
+            slotRangeArray *slots = slotRangeArrayDup(arg);
+            if (slotRangeArrayNormalizeAndValidate(slots, &errsds) != C_OK) {
+                slotRangeArrayFree(slots);
+                ret = C_ERR;
+                break;
+            }
+            ret = asmCreateImportTask(task_id, slots, &errsds) ? C_OK : C_ERR;
             break;
-        case ASM_EVENT_CANCEL:
+        }
+        case ASM_EVENT_CANCEL: {
             num_cancelled = clusterAsmCancel(task_id, "user request");
             if (arg) *((int *)arg) = num_cancelled;
             ret = C_OK;
             break;
-        case ASM_EVENT_HANDOFF:
+        }
+        case ASM_EVENT_HANDOFF: {
             ret = clusterAsmHandoff(task_id, &errsds);
             break;
-        case ASM_EVENT_DONE:
+        }
+        case ASM_EVENT_DONE: {
             ret = clusterAsmDone(task_id, &errsds);
             break;
-        default:
+        } default: {
```

Comment:
so default block should be like?
```suggestion
        }
        default: {
```